src/core/stdc/stdlib.d

```diff
@@ -159,6 +159,17 @@ else
 ///
 void*   malloc(size_t size) pure;
 ///
+void*   pureMalloc(size_t size) pure
+{
+    const errno = pureGetErrno();
+    typeof(return) ret = malloc(size);
+    cast(void)pureSetErrno(errno);
+    return ret;
+}
+extern (C) pure pragma(mangle, "getErrno") int pureGetErrno(); // for internal use
+extern (C) pure pragma(mangle, "setErrno") int pureSetErrno(int); // for internal use
```

Comment:
Also, since those functions are for internal use, why not make them `private` or `package`, or declare them inside `pureMalloc`? They should be used only in a controlled manner, and reviewed case-by-case.